File: 1408/StringMatchingInAnArray.py
# https://leetcode.com/problems/string-matching-in-an-array/description/?envType=daily-question&envId=2025-01-07

class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:

        def checkForSubstring(stringToCheck, currentString):
            lengthCurrentString=len(currentString)
            for i in range(0, lengthCurrentString, 1):
                tempString=""
                for j in range(i, lengthCurrentString, 1):
                    tempString+=currentString[j]
                    if tempString==stringToCheck:
                        return True
            return False

        resultantList=[]
        lengthOfWords=len(words)
        for i in range(0, lengthOfWords, 1):
            stringToCheck=words[i]
            for j in range(0, lengthOfWords, 1):
                if i!=j:
                    currentString=words[j]
                    if len(stringToCheck)<=len(currentString):
                        isSubstring=checkForSubstring(stringToCheck, currentString)
                        if isSubstring==True:
                            resultantList.append(stringToCheck)
                            break
        return resultantList

File: 1408/StringMatchingInAnArray.cpp
/*
https://leetcode.com/problems/string-matching-in-an-array/description/?envType=daily-question&envId=2025-01-07
*/

class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        vector<string>res;
        int sz=words.size();
        for(int i=0; i<sz; ++i){
            string curr=words[i];
            for(int j=0; j<sz; ++j){
                if(i!=j){
                    string next=words[j];
                    if(curr.length()<=next.length()){
                        auto ifpresent=next.find(curr);
                        if(ifpresent!=string::npos){
                            res.push_back(curr);
                            break;
                        }
                    }
                }
            }
        }
        return res;
    }
};